client/signup/config/step-components.js

```diff
@@ -1,193 +1,78 @@
-);
-const PlansStepComponent = props => (
-	<AsyncLoad require="signup/steps/plans" placeholder={ null } { ...props } />
-);
-const SiteComponent = props => (
-	<AsyncLoad require="signup/steps/site" placeholder={ null } { ...props } />
-);
-const RebrandCitiesWelcomeComponent = props => (
-	<AsyncLoad require="signup/steps/rebrand-cities-welcome" placeholder={ null } { ...props } />
-);
-const RewindMigrate = props => (
-	<AsyncLoad require="signup/steps/rewind-migrate" placeholder={ null } { ...props } />
-);
-const RewindWereBacking = props => (
-	<AsyncLoad require="signup/steps/rewind-were-backing" placeholder={ null } { ...props } />
-);
-const RewindAddCreds = props => (
-	<AsyncLoad require="signup/steps/rewind-add-creds" placeholder={ null } { ...props } />
-);
-const RewindFormCreds = props => (
-	<AsyncLoad require="signup/steps/rewind-form-creds" placeholder={ null } { ...props } />
-);
-const SiteOrDomainComponent = props => (
-	<AsyncLoad require="signup/steps/site-or-domain" placeholder={ null } { ...props } />
-);
-const SitePicker = props => (
-	<AsyncLoad require="signup/steps/site-picker" placeholder={ null } { ...props } />
-);
-const SiteTitleComponent = props => (
-	<AsyncLoad require="signup/steps/site-title" placeholder={ null } { ...props } />
-);
-const SiteStyleComponent = props => (
-	<AsyncLoad require="signup/steps/site-style" placeholder={ null } { ...props } />
-);
-const SiteTopicComponent = props => (
-	<AsyncLoad require="signup/steps/site-topic" placeholder={ null } { ...props } />
-);
-const SiteTypeComponent = props => (
-	<AsyncLoad require="signup/steps/site-type" placeholder={ null } { ...props } />
-);
-const SiteInformationComponent = props => (
-	<AsyncLoad require="signup/steps/site-information" placeholder={ null } { ...props } />
-);
-const SurveyStepComponent = props => (
-	<AsyncLoad require="signup/steps/survey" placeholder={ null } { ...props } />
-);
-const ThemeSelectionComponent = props => (
-	<AsyncLoad require="signup/steps/theme-selection" placeholder={ null } { ...props } />
-);
-const UserSignupComponent = props => (
-	<AsyncLoad require="signup/steps/user" placeholder={ null } { ...props } />
-);
-const PlansStepWithoutFreePlan = props => (
-	<AsyncLoad require="signup/steps/plans-without-free" placeholder={ null } { ...props } />
-);
-const PlansAtomicStoreComponent = props => (
-	<AsyncLoad require="signup/steps/plans-atomic-store" placeholder={ null } { ...props } />
-);
-const ReaderLandingStepComponent = props => (
-	<AsyncLoad require="signup/steps/reader-landing" placeholder={ null } { ...props } />
-);
-const LaunchSiteComponent = props => (
-	<AsyncLoad require="signup/steps/launch-site" placeholder={ null } { ...props } />
+const loadStep = memoize( stepName =>
+	React.lazy( () =>
+		import( /* webpackChunkName: "async-load-signup-steps-[request]", webpackInclude: /signup\/steps\/[a-z-]+\/index.jsx$/ */ `signup/steps/${ stepName }` )
```

Comment:
This line is interesting. I didn't realise you could generate a chunk for each one of these but keep them all in their own chunks. Great work! 👍

Reply:
We already do something similar for Moment.js locales, where each locale file is bundled into its own chunks.

Packaging whole modules is a bit tricky: I need to specify the full path to the index file of each module. Directories are not checked against the `webpackInclude` regexp, they are only traversed and descended into.